api/src/main/java/org/pmiops/workbench/api/OfflineTestUsersController.java

```diff
@@ -0,0 +1,60 @@
+package org.pmiops.workbench.api;
+
+import static org.pmiops.workbench.firecloud.IntegrationTestUsers.COMPLIANT_USER;
+
+import java.util.logging.Logger;
+import javax.inject.Provider;
+import org.pmiops.workbench.config.WorkbenchConfig;
+import org.pmiops.workbench.impersonation.ImpersonatedUserService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class OfflineTestUsersController implements OfflineTestUsersApiDelegate {
+  private static final Logger LOGGER = Logger.getLogger(OfflineTestUsersController.class.getName());
+
+  private final Provider<WorkbenchConfig> workbenchConfigProvider;
+  private final ImpersonatedUserService impersonatedUserService;
+
+  @Autowired
+  public OfflineTestUsersController(
+      Provider<WorkbenchConfig> workbenchConfigProvider,
+      ImpersonatedUserService impersonatedUserService) {
+    this.workbenchConfigProvider = workbenchConfigProvider;
+    this.impersonatedUserService = impersonatedUserService;
+  }
+
+  public ResponseEntity<Void> ensureTestUserTosCompliance() {
+    WorkbenchConfig config = workbenchConfigProvider.get();
+
+    // TODO: is there a better way to check for when we're executing in the test env?
+    if (config.server.projectId.equals("all-of-us-workbench-test")) {
```

Comment:
all-of-us-workbench-test should this be a constant

Reply:
changed my approach here - now checks if the user is in the appropriate gsuite domain